src/sagemaker/jumpstart/types.py

```diff
@@ -736,6 +736,334 @@ def _get_regional_property(
         return alias_value
 
 
+class JumpStartBenchmarkStat(JumpStartDataHolderType):
+    """Data class JumpStart benchmark stats."""
+
+    __slots__ = ["benchmark_name", "instance_type", "latency_ms"]
+
+    def __init__(self, spec: Dict[str, Any]):
+        """Initializes a JumpStartBenchmarkStat object
+
+        Args:
+            spec (Dict[str, Any]): Dictionary representation of benchmark stat.
+        """
+        self.from_json(spec)
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of benchmark stats.
+        """
+        self.benchmark_name: str = json_obj["benchmark_name"]
+        self.instance_type: str = json_obj["instance_type"]
+        self.latency_ms: int = json_obj["latency_ms"]
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartBenchmarkStat object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+
+class JumpStartInferenceComponent(JumpStartDataHolderType):
```

Comment:
can we rename this `JumpStartMetadataComponent`, and have 1 type for training and inference?

`InferenceComponent` could get confused with goldfinch

Reply:
the idea with these classes is to be able to override _every_ field of the spec, with the exception of some (hard code the exception list). And ideally this parity would be maintained automatically.  Can yo refactor `JumpStartModelSpecs` to inherit another class/interface `JumpStartRawMetadataFields` that has all the fields, and then have `class JumpStartMetadataComponent(JumpStartRawMetadataFields)` inherit the class?

Reply:
Makes a lot of sense, I was under the impression that we are only overriding a subset of the specs as currently in this component. I'll refactor

Reply:
no worries!

Reply:
@evakravi @Captainia let's please not use "raw metadata" to mean anything outside of the contributor-side config. Otherwise, the terminology becomes very confusing. `JumpstartMetadataFields` should be fine.

We should also align on the overall approach. Here are my thoughts:

1. Each "config_name" from the design is called a `Preset`
2. `JumpStartModelSpecs` contains a set of `Presets`, but uses the default one for its initialization of fields
3.  `Presets` are constructed from an array of `ConfigSets` (i.e. `component` -> `ConfigSet`)
4. No distinction between `InferenceConfigSet` or `TrainingConfigSet`, there is just `ConfigSet`
5. `ConfigSet` contains all the JumpStart fields